envoy-filters/mgw-source/filters/http/mgw-wasm-websocket/filter.cc

```diff
@@ -208,26 +230,14 @@ FilterDataStatus MgwWebSocketContext::onResponseBody(size_t body_buffer_length,
     // Read ext_authz_metadata_ metdata saved as a member variable
     *request.mutable_metadata() = *this->metadata_;
     request.set_payload(std::string(body->view()));
+    request.set_direction(WebSocketFrameRequest_MessageDirection_SUBSCRIBE);
 
     // Perform throttling logic.
     // If the throttle state is underlimit and if the gRPC stream is open, send WebSocketFrameRequest. 
     // If no gRPC stream, try to open a new stream and then send. 
     if(this->throttle_state_ == ThrottleState::UnderLimit){
-      if(this->handler_state_ == HandlerState::OK){
-        LOG_TRACE(std::string("gRPC bidi stream available. publishing frame data..."));
-        auto ack = this->stream_handler_->send(request, false);
-        if (ack != WasmResult::Ok) {
-          LOG_TRACE(std::string("error sending frame data")+ toString(ack));
-        }
-        LOG_TRACE(std::string("frame data successfully sent:"+ toString(ack)));
-      }else{
-        establishNewStream();
-        auto ack = this->stream_handler_->send(request, false);
-        if (ack != WasmResult::Ok) {
-          LOG_TRACE(std::string("error sending frame data")+ toString(ack));
-        }
-        LOG_TRACE(std::string("frame data successfully sent:"+ toString(ack)));
-      }
+      request.set_apim_error_code(0);
+      sendEnforcerRequest(this, request);
```

Comment:
Try sending the enforcerRequest when the throttleState is in failureModeallowed/ blocked